src/internal/connector/onedrive/api/mock/drive.go

```diff
@@ -0,0 +1,56 @@
+package mock
+
+import (
+	"context"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/alcionai/corso/src/internal/connector/graph/api"
+)
+
+type PageLinker struct {
+	Link *string
+}
+
+func (pl *PageLinker) GetOdataNextLink() *string {
+	return pl.Link
+}
+
+type PagerResult struct {
+	Drives   []models.Driveable
+	NextLink *string
+	Err      error
+}
+
+type DrivePager struct {
+	ToReturn []PagerResult
+	GetIdx   int
+}
+
+func (p *DrivePager) GetPage(context.Context) (api.PageLinker, error) {
+	if len(p.ToReturn) <= p.GetIdx {
+		return nil, assert.AnError
```

Comment:
nit: for debugability maybe we should wrap this with a message of "out of bounds" or something? That will also give stack trace info. Or just return a clues error and forget the wrap call?
```suggestion
		return nil, clues.Wrap(assert.AnError, "index out of bounds")
```

Reply:
Sure, why not?